src/solid/lib/base.ts

```diff
@@ -0,0 +1,187 @@
+import {
+  clearNode,
+  createNode,
+  createWatch,
+  is,
+  Scope,
+  scopeBind,
+  step,
+  Store,
+  Unit,
+} from 'effector'
+import {throwError} from './throw'
+import {Accessor, batch, createMemo, createSignal, onCleanup} from 'solid-js'
+
+const stateReader = <T>(store: Store<T>, scope?: Scope) =>
+  scope ? scope.getState(store) : store.getState()
+const basicUpdateFilter = <T>(upd: T, oldValue: T) => upd !== oldValue
+
+export function useUnitBase<Shape extends {[key: any]: Unit<any>}>(
+  shape: Shape,
+  scope?: Scope,
+) {
+  const isShape = !is.unit(shape) && typeof shape === 'object'
+  const normShape = isShape ? shape : {unit: shape}
+  const isList = Array.isArray(normShape)
+  const entries = Object.entries(normShape)
+
+  const initial = isList ? [] : {}
+  const stores = []
+  const storeIdMap = {}
+  const storeSetterMap = {}
+
+  for (const [key, value] of entries) {
+    if (is.store(value)) {
+      if (stores.includes(value)) {
+        throwError(`useUnit store at key "${key}" is already exists in shape`)
+      }
+      stores.push(value)
+      storeIdMap[value.graphite.id] = key
+      const [get, set] = createSignal(stateReader(value, scope))
+      initial[key] = get
+      storeSetterMap[key] = set
+    } else {
+      if (!is.unit(value)) throwError('expected useUnit argument to be a unit')
+      initial[key] = scope ? scopeBind(value, {scope}) : value
+    }
+  }
+
+  const seq = [
+    step.compute({priority: 'effect'}),
+    step.compute({priority: 'sampler', batch: true}),
+    step.run({
+      fn: () => {
+        batch(() => {
+          for (const store of stores) {
+            const key = storeIdMap[store.graphite.id]
+            storeSetterMap[key](() => store.getState())
```

Comment:
I think, it is better to use `stateReader` here 🤔 
```suggestion
            storeSetterMap[key](() => stateReader(store, scope))
```
so it explicitly reads from scope, when its needed

AFAIK, `store.getState()` actually will read value from scope, if called in its context, so current code will also work correctly, but since it is not a documented public feature, i think it sould be better to read from scope explicitly